ci-operator/step-registry/openstack-k8s-operators/gather/openstack-k8s-operators-gather-commands.sh

```diff
@@ -1,21 +1,64 @@
 #!/usr/bin/env bash
 
 set -x
+set +e
+
+BASE_DIR=${HOME:-"/alabama"}
+MUST_GATHER_TIMEOUT=${MUST_GATHER_TIMEOUT:-"5m"}
+
+function create_out_dir {
+  local OUT_DIR=${1}
+  if [ ! -d "./${OUT_DIR}" ]; then
+    mkdir ${OUT_DIR}
+  fi
+}
+
+function gen_pods_debug_cmd {
+  create_out_dir ${1}
+  local cmd=("echo 'Logs for pod: {}'; oc describe pod {} > ${1}/{}-describe.log; oc logs --all-containers=true {} >> ${1}/{}-containers.log")
+  echo "${cmd[@]}"
+}
+
+function gen_api_resources_debug_cmd {
+  create_out_dir ${1}
+  local cmd=("echo 'Info for api-resource: {}'; oc get {} -o yaml > ${1}/{}.log")
+  echo "${cmd[@]}"
+}
 
 # We don't want to use OpenShift-CI build cluster namespace
 unset NAMESPACE
-
 oc project openstack
 
-oc logs ceph
+pushd "${BASE_DIR}" || exit
+mkdir logs
+pushd logs || exit
 
-oc get OpenStackControlPlane openstack -o json
+oc get all > get_all.log
 
-oc get all
+CMD=$(gen_api_resources_debug_cmd "resources")
+oc api-resources | grep 'openstack' | awk '{print $1}' | xargs -n1 -I {} sh -c "${CMD[@]}"
+# RabbitMQCluster
+oc get -o yaml RabbitMQCluster > RabbitMQCluster.log
 
-oc get -o yaml MariaDB,RabbitMQCluster,KeystoneAPI,PlacementAPI,Glance,Cinder,NeutronAPI,Nova
+CMD=$(gen_pods_debug_cmd "controller-manager")
+oc get pods -n openstack --show-labels | grep -i '.*control\-plane\=.*controller\-manager.*' | awk '{print $1}' | \
+xargs -n1 -I {} sh -c "${CMD[@]}"
 
-oc get pods -n openstack --show-labels | grep -i '.*control\-plane\=.*controller\-manager.*' | awk '{print $1}' | xargs -n1 oc logs
+CMD=$(gen_pods_debug_cmd "services")
+oc get pods -n openstack --no-headers | egrep -iv controller | awk '{print $1}' | xargs -n1 -I {} sh -c "${CMD[@]}"
 
+CMD=$(gen_pods_debug_cmd "errors")
 # Capture logs from service containers if container is not in Running or Completed state
-oc get pods -n openstack --no-headers | egrep -iv controller | egrep -iv 'Running|Completed' | awk '{print $1}' | xargs -n1 -I {} sh -c 'echo "Logs for pod: {}"; oc describe pod {}; oc logs --all-containers=true {}'
+oc get pods -n openstack --no-headers | egrep -iv controller | egrep -iv 'Running|Completed' | awk '{print $1}' | \
```

Comment:
For Openshift core operators, pods information [1] is collected in must gather itself but looks like this is not available for OLM operators reading[2]  and [3].

Looks like we need to manually collect pods information out of must-gather.

[1] https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/37084/rehearse-37084-pull-ci-openstack-k8s-operators-openstack-operator-master-openstack-operator-build-deploy-tempest/1633970107900235776/artifacts/openstack-operator-build-deploy-tempest/openstack-k8s-operators-gather/artifacts/must-gather/quay-io-openshift-release-dev-ocp-v4-0-art-dev-sha256-0291e5ed752e993177e25c1291484b63bc5ab38bd8d6f0f8020d78c8bc6cd5f6/namespaces/openshift-dns/pods/

[2] https://github.com/openshift/must-gather/issues/319
[3] https://github.com/openshift/must-gather/issues/247